app/celery/v3/notification_tasks.py

```diff
@@ -101,35 +103,41 @@ def v3_process_notification(request_data: dict, service_id: str, api_key_id: str
             template = reader_session.execute(query).one().Template
             notification.template_version = template.version
         except (MultipleResultsFound, NoResultFound):
+            notification.status = NOTIFICATION_PERMANENT_FAILURE
             notification.status_reason = "The template does not exist."
```

Comment:
We should split this into separate "except" blocks because only NoResultFound indicates a non-existent template.  MultipleResultsFound indicates a database integrity issue.

Reply:
done